src/compiler/checker.ts

```diff
@@ -6814,6 +6904,22 @@ module ts {
             return undefinedType;
         }
 
+        function checkAwaitExpression(node: AwaitExpression): Type {
+            // Grammar checking
+            if (!(node.parserContextFlags & ParserContextFlags.Await)) {
+                var parameter = getContainingParameter(node);
+                if (parameter && parameter.parserContextFlags & ParserContextFlags.Await) {
+                    grammarErrorAfterFirstToken(node, Diagnostics._0_expression_is_not_allowed_in_an_initializer, "await");

```

Comment:
I used the placeholder as I also use this message for yield.
________________________________
From: Jason Freeman<mailto:notifications@github.com>
Sent: ý2/ý13/ý2015 6:04 PM
To: Microsoft/TypeScript<mailto:TypeScript@noreply.github.com>
Cc: Ron Buckton<mailto:Ron.Buckton@microsoft.com>
Subject: Re: [TypeScript] Async Functions (#1781)


In src/compiler/checker.ts<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24710784>:

> @@ -6814,6 +6904,22 @@ module ts {
>              return undefinedType;
>          }
>
> +        function checkAwaitExpression(node: AwaitExpression): Type {
> +            // Grammar checking
> +            if (!(node.parserContextFlags & ParserContextFlags.Await)) {
> +                var parameter = getContainingParameter(node);
> +                if (parameter && parameter.parserContextFlags & ParserContextFlags.Await) {
> +                    grammarErrorAfterFirstToken(node, Diagnostics._0_expression_is_not_allowed_in_an_initializer, "await");


Also, I would just bake "await" into the message instead of passing it as an argument.

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24710784>.